src/python/pants/bin/local_pants_runner.py

```diff
@@ -156,10 +159,18 @@ def create(
             cancellation_latch,
         )
 
+        (git_binary,) = graph_session.scheduler_session.product_request(
+            GitBinary, [Params(GitBinaryRequest())]
+        )
+        (maybe_git_worktree,) = graph_session.scheduler_session.product_request(
+            MaybeGitWorktree, [Params(GitWorktreeRequest(), git_binary)]
+        )
+
         specs = calculate_specs(
```

Comment:
This is the right first step based on our conversation. But yea, in a second commit in this PR, it will be time to try the actual rule graph change described in point 3 here: #13757.